galley/pkg/server/settings/args.go

```diff
@@ -51,6 +53,12 @@ type Args struct { // nolint:maligned
 	// KubeInterface has an already created K8S interface, will be reused instead of creating a new one
 	KubeInterface *kubernetes.Clientset
 
+	// InsecureGRPC is an existing GRPC server, will be used by Galley instead of creating its own
+	InsecureGRPC *grpc.Server
```

Comment:
+1. @costinm what is the thought process here? Is this to allow tests to manually set the server?

Reply:
Args are params passed in to start Galley. 

This includes a kube client to use, as well as a grpc.Server - it is not for manual test, we want to use a single K8S client and grpc server for Pilot, Galley, Citadel instead of each component creating its own. 

Regarding the mess - as soon as we get istiod working we can clean up the duplicated code. 

But we certainly don't want to continue to have each component create http and grpc servers and dealing with certs.